### PR TITLE
Add GasEstimate FFCAPI implementation

### PR DESCRIPTION
**Pull request by matthew1001**
This PR implements the `GasEstimate` FFCAPI method added to `FireFly Transaction Manager` release 1.2.13 (https://github.com/hyperledger/firefly-transaction-manager/releases/tag/v1.2.13)

Most of the logic already existed for internal use in `TransactionPrepare` so this PR primarily just exposes it as an FFCAPI endpoint.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,9 +4,9 @@ go 1.19
 
 require (
 	github.com/hashicorp/golang-lru v0.5.4
-	github.com/hyperledger/firefly-common v1.2.10
-	github.com/hyperledger/firefly-signer v1.1.7
-	github.com/hyperledger/firefly-transaction-manager v1.2.11
+	github.com/hyperledger/firefly-common v1.2.11
```
Comment (peterbroadhurst) [inline go.mod:7]: noting I think this now supersedes https://github.com/hyperledger/firefly-evmconnect/pull/83

File: internal/ethereum/exec_query.go [hunk 2]
```diff
@@ -125,19 +125,22 @@ func (c *ethConnector) callTransaction(ctx context.Context, tx *ethsigner.Transa
...
-	var jsonData []byte
-	outputValueTree, err := method.Outputs.DecodeABIDataCtx(ctx, outputData, 0)
-	if err == nil {
-		// Serialize down to JSON, and wrap in a JSONAny
-		jsonData, err = c.serializer.SerializeJSONCtx(ctx, outputValueTree)
-	}
-	if err != nil {
-		log.L(ctx).Warnf("Invalid return data: %s", outputData)
-		return nil, "", i18n.NewError(ctx, msgs.MsgReturnDataInvalid, err)
+	if method != nil {
```
Comment (peterbroadhurst) [inline internal/ethereum/exec_query.go:128]: This allows us to handle Gas estimation and Call requests for simple transfers, right?

**Review by peterbroadhurst — APPROVED** [review]
Looks good. Just wanted to validate the intent of one of the code changes, as I didn't spot it called out in the PR description

File: internal/ethereum/exec_query.go (hunk 2 above)
Comment (matthew1001) [reply internal/ethereum/exec_query.go:128]: Yeah that's exactly right. Should have spelled out the reason for it, but yeah that's it.

**Pull request merged**